lib/inspec/base_cli.rb

```diff
@@ -71,24 +73,70 @@ def self.exec_options
         desc: 'Write out a lockfile based on this execution (unless one already exists)'
       option :backend_cache, type: :boolean,
         desc: 'Allow caching for backend command output.'
+      option :show_progress, type: :boolean,
+        desc: 'Show progress while executing tests.'
     end
 
     def self.default_options
       {
         exec: {
+          'reporter' => {
+            'cli' => nil,
+          },
+          'show_progress' => false,
           'color' => true,
           'create_lockfile' => true,
           'backend_cache' => false,
         },
       }
     end
 
+    def self.clean_reporters(opts)
+      # Parse report options from CLI
+      if opts['reporter'].is_a?(Array)
+        stdout = 0
+        reports = {}
+        opts['reporter'].each do |report|
+          k, v = report.split(':')
+          v = nil if v == '-'
+          reports[k] = v
+          stdout += 1 if v.nil?
+        end
+
+        raise ArgumentError, 'The option --reporter can only have a single report outputting to stdout.' if stdout > 1
+
+        opts['reporter'] = reports
+      end
+
+      # Catch anyone using the legacy --format option
+      unless opts['format'].nil?
+        opts['reporter'] = { opts['format'] => nil }
+        opts.delete('format')
+      end
+
+      # Catch any dynamic inspec runners or tests
+      opts['reporter'] = { 'cli' => nil } if opts['reporter'].nil?
+
+      opts
+    end
+
     private
 
+    def suppress_log_output?(opts)
+      match = %w{json json-min json-rspec junit} & opts['reporter'].keys
```

Comment:
Nope. I also wonder if we're making this hard on ourselves and shouldn't just change the default log destination to STDERR instead of STDOUT. If we do that, all of this selective suppression can go away because people can just redirect STDOUT to a file if they care without mucking with the reporter output. Even though it's technically a breaking change, part of me thinks that outputting log output to the same stream as formatted data is a bug, and I'd be happy to have us change this without requiring a major version bump. I don't think the log output is consumed by many except when debugging something.